src/server/injected/injectedScript.ts

```diff
@@ -662,7 +662,12 @@ export class InjectedScript {
 
   isElementDisabled(element: Element): boolean {
     const elementOrButton = element.closest('button, [role=button]') || element;
-    return ['BUTTON', 'INPUT', 'SELECT', 'TEXTAREA'].includes(elementOrButton.nodeName) && elementOrButton.hasAttribute('disabled');
+    if (['BUTTON', 'INPUT', 'SELECT', 'TEXTAREA'].includes(elementOrButton.nodeName)) {
+      return elementOrButton.hasAttribute('disabled');
+    } else {
+      const ariaDisabled = elementOrButton.getAttribute('aria-disabled');
```

Comment:
The logic for aria-disabled is actually more complicated than this. If the element is focusable, the disabled state can be inherited from parent elements.
See https://www.w3.org/TR/wai-aria-1.1/#aria-disabled and https://source.chromium.org/chromium/chromium/src/+/master:third_party/blink/renderer/modules/accessibility/ax_object.cc;l=3171;drc=6cdb24a4ce9d4151035c1f133833137d2e2881d1

Reply:
Can `elementOrButton.closest('[aria-disabled]')` lookup be sufficient?

Reply:
sorry for the late reply. The logic here would actually be more complicated, because you have to first check if the element takes focus, and only then do `element.closest`. I think its fine to do what you did and differ from the aria spec here. We can fix edge cases if people complain.